tools/scripts/features/frcnn/extract_features_frcnn.py

```diff
@@ -0,0 +1,256 @@
+        parser.add_argument("--batch_size", type=int, default=2, help="Batch size")
+        parser.add_argument(
+            "--output_folder", type=str, default="./output", help="Output folder"
+        )
+        parser.add_argument("--image_dir", type=str, help="Image directory or file")
+        parser.add_argument(
+            "--feature_index",
+            type=int,
+            help="The index of the feature to extract",
+            default=None,
+        )
+        parser.add_argument(
+            "--exclude_list",
+            type=str,
+            help="List of images to be excluded from feature conversion. "
+            + "Each image on a new line",
+            default="./list",
+        )
+        parser.add_argument(
+            "--confidence_threshold",
+            type=float,
+            default=0,
+            help="Threshold of detection confidence above which boxes will be selected",
+        )
+        # TODO finish background flag
+        parser.add_argument(
+            "--background",
+            action="store_true",
+            help="The model will output predictions for the background class when set",
+        )
+        return parser
+
+    def _build_detection_model(self):
+        if self.args.config_file:
+            frcnn_cfg = Config.from_pretrained(self.args.config_file)
+        else:
+            frcnn_cfg = Config.from_pretrained(
+                self.MODEL_URL.get(self.args.model_name, self.args.model_name)
+            )
+        if self.args.model_file:
+            frcnn = GeneralizedRCNN.from_pretrained(
+                self.args.model_file, config=frcnn_cfg
+            )
+        else:
+            frcnn = GeneralizedRCNN.from_pretrained(
+                self.MODEL_URL.get(self.args.model_name, self.args.model_name),
+                config=frcnn_cfg,
+            )
+
+        return frcnn, frcnn_cfg
+
+    def get_frcnn_features(self, image_paths):
+        image_preprocess = Preprocess(self.frcnn_cfg)
+
+        images, sizes, scales_yx = image_preprocess(image_paths)
+
+        output_dict = self.frcnn(
+            images,
+            sizes,
+            scales_yx=scales_yx,
+            padding="max_detections",
+            max_detections=self.frcnn_cfg.max_detections,
+            return_tensors="pt",
+        )
+
+        return output_dict
+
+    def _save_feature(self, file_name, feature):
+        file_base_name = os.path.basename(file_name)
+        file_base_name = file_base_name.split(".")[0]
+        file_base_name = file_base_name + ".npy"
+        np.save(os.path.join(self.args.output_folder, file_base_name), feature)
+
+    def _process_features(self, features, index):
+        feature_keys = [
+            "obj_ids",
+            "obj_probs",
+            "attr_ids",
+            "attr_probs",
+            "boxes",
+            "sizes",
+            "preds_per_image",
+            "roi_features",
+            "normalized_boxes",
+        ]
```

Comment:
We also need to make sure here if the name of the fields saved for these features are same as the name of the fields saved by `extract_features_vmb.py` features. That will ensure we can load the features and train the models without any hassle.